tutorial/Execution.md

```diff
@@ -0,0 +1,412 @@
+
+There's also a companion object `Execution` that enables us get the `Config` or `Mode`. A use of `Execution` might look like:
+
+```scala
+val config: Config = // ...
+val mode: Mode = // ...
+Execution.getConfig.run(config, mode)
+```
+
+Not very useful yet.
+
+### Map
+
+Just like scala's `Seq`, `Option`, `Future`, etc., `Execution` can be *mapped over* with its `map` method. In the context of `Seq`, calling the `map` method will transform every element into a new element via some function. In `Execution`, the `map` method gives us a new `Execution` that will produce the value of the old `Execution` and then transform that value with the function. Remember, `Execution` is a plan to do something; it won't be run until we call the `run` method:
+
+```scala
+case class Execution[T](private doSomething: (Config, Mode) => T) {
+  final def map[U](transform: T => U): Execution[U] = {
+    def newDoSomething(config: Config, mode: Mode): U = {
+      val result = doSomething(config, mode))
+      transform(result)
+    }
+
+    Execution(newDoSomething)
+  }
+
+  // ... other previously defined methods ...
+}
+```
+
+This is pretty powerful. `Execution` can now be used to access anything in the `Config` or `Mode`, or even do arbitrary work:
+
+```scala
+val config: Config = // ...
+val mode: Mode = // ...
+
+val config: ConfigFunProperty = Execution
+  .getConfig
+  .map { config => config.get("fun") }
+  .run(config, mode)
+
+val fivePlusThree = Execution
+  .getMode
+  .map { _ => 5 + 3 }
+```
+
+### From
+
+*Lifting* arbitrary code into `Execution` is used so frequent that we're going to add a helper method, `from`, for it:
+
+```
+object Execution {
+  final def from[T](value: => T): Execution[U] = {
+    def newDoSomething(config: Config, mode: Mode) = value
+    Execution(newDoSomething)
+  }
+
+  // ... other previously defined methods ...
+}
+```
+
+Why is it `value: => T` and not `value: T`? This scala syntax here `=>` is "call by name" and is essentially receiving a function of 0 parameters. Check out the scala documentation for more.
+
+At this point, we can write an `Execution` that reads from a service and uses its result to do something:
+
+```scala
+val config: Config = // ...
+val mode: Mode = // ...
+val service: String => String = // ...
+
+val callOutToService = Execution.from {
+  service("eat galaxies")
+}
+
+val reversedServiceResult = callOutToService.map { serviceResult =>
+  serviceResult.reverse
+}
+
+reversedServiceResult.run(config, mode)
+```
+
+### Running a Scalding job
+
+Writing a single Scalding job with `TypedPipe` and `Grouped` will not be covered in this tutorial; however, in order to run Scalding jobs with `Execution`, we need to be familiar with with 2 methods on `TypedPipe`: `writeExecution` and `toIterableExecution`:
+
+```scala
+val config: Config = // ...
+val mode: Mode = // ...
+
+val source: TypedSource[Int] = // ...
```

Comment:
I think (here and elsewhere) examples are easier to follow when you don't use primitive types. Eg, make this:

`TypedPipe[User]` and `val getAddress = u: User => u.address` etc. (and use `Address` instead of String for address). No need to actually define the User and Address types tho.